src/cron/personalizedDigest.ts

```diff
@@ -1,32 +1,44 @@
+import { addHours, startOfHour } from 'date-fns';
 import {
   DEFAULT_TIMEZONE,
   getPersonalizedDigestPreviousSendDate,
-  getPersonalizedDigestSendDate,
+  digestPreferredHourOffset,
   notifyGeneratePersonalizedDigest,
   schedulePersonalizedDigestSubscriptions,
 } from '../common';
 import {
   User,
   UserPersonalizedDigest,
   UserPersonalizedDigestSendType,
+  UserPersonalizedDigestType,
 } from '../entity';
 import { Cron } from './cron';
 import { Brackets } from 'typeorm';
 
 const sendType = UserPersonalizedDigestSendType.weekly;
+const digestTypes = [UserPersonalizedDigestType.Digest];
 
 const cron: Cron = {
   name: 'personalized-digest',
   handler: async (con, logger) => {
-    const nextPreferredDay = (new Date().getDay() + 1) % 7;
     const personalizedDigestQuery = con
       .createQueryBuilder()
       .select('upd.*, u.timezone')
       .from(UserPersonalizedDigest, 'upd')
       .leftJoin(User, 'u', 'u.id = upd."userId"')
-      .where('upd."preferredDay" = :nextPreferredDay', {
-        nextPreferredDay,
-      })
+      .where(
+        `(EXTRACT(DAY FROM NOW() AT TIME ZONE COALESCE(NULLIF(u.timezone, ''), '${DEFAULT_TIMEZONE}')) - 1) = upd."preferredDay"`,
+        {
+          defaultTimezone: DEFAULT_TIMEZONE,
+        },
+      )
```

Comment:
basically similar how we are guessing the hour in user's timezone now we are also guessing the day of week in the timezone so we know when we need to send

`-1` is here because psql weekdays are 1-7

Reply:
I think you took `day` (1-31) instead of `dow` (0-6)
https://www.postgresql.org/docs/8.1/functions-datetime.html

It probably looked ok because it's only the 5th now.

Reply:
damn you are righh 100%, this is why we do code reviews, tnx! fixed in f4998457b2ab80cb23515f74bd133790a18f32ea